kola/tests/misc/tang.go

```diff
@@ -0,0 +1,300 @@
+	// root partition, which is vda9 in the QEMU disk image.
+	IgnitionConfigRootTang = `{
+		"ignition": {
+			"config": {},
+			"timeouts": {},
+			"version": "3.3.0"
+		},
+		"networkd": {},
+		"storage": {
+			"disks": [
+				{
+					"device": "/dev/disk/by-id/virtio-secondary",
+					"wipeTable": true,
+					"partitions": [
+						{
+							"label": "root",
+							"number": 1
+						}
+					]
+				}
+			],
+			"luks": [
+				{
+					"name": "rootencrypted",
+					"device": "/dev/disk/by-partlabel/root",
+					"clevis": {
+						"tang": [
+							{
+								"url": "http://{{ .TangIP }}:{{ .TangPort }}",
+								"thumbprint": "HkwVNDeKhzaVqWhXtXwEIGNILRZt4cBWWb0kI1-a0NM"
+							}
+						]
+					}
+				}
+			],
+			"filesystems": [
+				{
+					"device": "/dev/disk/by-id/dm-name-rootencrypted",
+					"format": "ext4",
+					"label": "ROOT"
+				}
+			]
+		},
+		"systemd": {
+			"units": [{
+			"name": "remove-root-label.service",
+			"enabled": true,
+			"contents": "[Service]\nType=oneshot\nExecStart=wipefs -a /dev/vda9\n[Install]\nWantedBy=multi-user.target"
+			}]
+		}
+	}`
+
+	IgnitionConfigNonRootTang = `{
+		"ignition": {
+			"config": {},
+			"timeouts": {},
+			"version": "3.3.0"
+		},
+		"networkd": {},
+		"storage": {
+			"disks": [
+				{
+					"device": "/dev/disk/by-id/virtio-secondary",
+					"wipeTable": true,
+					"partitions": [
+						{
+							"label": "data",
+							"number": 1
+						}
+					]
+				}
+			],
+			"luks": [
+				{
+					"name": "dataencrypted",
+					"device": "/dev/disk/by-partlabel/data",
+					"clevis": {
+						"tang": [
+							{
+								"url": "http://{{ .TangIP }}:{{ .TangPort }}",
+								"thumbprint": "HkwVNDeKhzaVqWhXtXwEIGNILRZt4cBWWb0kI1-a0NM"
+							}
+						]
+					}
+				}
+			],
+			"filesystems": [
+				{
+					"device": "/dev/disk/by-id/dm-name-dataencrypted",
+					"format": "ext4",
+					"label": "DATA",
+					"path": "/mnt/data"
+				}
+			]
+		},
+		"systemd": {
+			"units": [{
+			"name": "mnt-data.mount",
+			"enabled": true,
+			"contents": "[Mount]\nWhat=/dev/disk/by-label/DATA\nWhere=/mnt/data\nType=ext4\n\n[Install]\nWantedBy=local-fs.target"
+			}]
+		}
+	}`
+)
+
+func init() {
+	tangPort := 8007
+	// TODO: I let the Tang server bind to an IP of a networking interface in the root networking namespace. This works because
+	// traffic from within the VM to that IP will be routed to the host. However, this approach introduces an unappealing dependency
+	// between this test and how the network is setup for the VMs, and also it might bind Tang to an interface with a public IP.
+	// An alternative approach would be to add another TAP interface to the bridge and let the Tang server bind there, but that would
+	// require the Tang setup to happen outside of these tests and introducing more complexity in different parts of the code base.
+	// I'll decide whether to rewrite this or leave it as it is based on feedback by reviewers.
```

Comment:
With `Platforms:   []string{"qemu"},` that seems ok

Reply:
Okay, I removed the `TODO` comment.